### ISSUE TITLE
Incorrect "name[casing]" warning in ansible-lint 6.9.1

### ISSUE THREAD

**Issue by andy-maier** [ansible/ansible-lint#2789]
##### Summary

ansible-lint 6.9.1 raises a new "name[casing]" warning that was not raised before (e.g. in 6.9.0). The warning is raised on just some occurrences of the same task name pattern, and I was not able to generate a small test case that reproduces this.

Here is the output of one of our large playbooks, which includes several roles:

```
ansible-lint playbooks/partition_create.yml
WARNING: ansible-lint is no longer tested under Python 3.8 and will soon require 3.9. Do not report bugs for this version.
WARNING  Offline mode ignored because it is not supported by ansible versions before 2.14.
WARNING  Unable to resolve FQCN for module zhmc_nic
[WARNING]: Falling back to Ansible unique filter as Jinja2 one failed: 'ips' is undefined
WARNING  Unable to resolve FQCN for module zhmc_partition
WARNING  Unable to resolve FQCN for module zhmc_storage_group
WARNING  Unable to resolve FQCN for module zhmc_storage_volume
WARNING  Unable to resolve FQCN for module zhmc_storage_group_attachment
WARNING  Listing 2 violation(s) that are fatal
name[casing]: All names should start with an uppercase letter. (warning)
playbooks/roles/zoning/tasks/main.yml:137 Task/Handler: [SAN1] Ensure the 8 zones ZONE_{{ params.name | regex_replace('-', '_') }}_0x exist on SAN1

name[casing]: All names should start with an uppercase letter. (warning)
playbooks/roles/zoning/tasks/main.yml:155 Task/Handler: [SAN2] Ensure the 8 zones ZONE_{{ params.name | regex_replace('-', '_') }}_0x exist on SAN2


              Rule Violation Summary              
 count tag          profile  rule associated tags 
     2 name[casing] moderate idiom (warning)      

Passed with basic profile, 1/5 star rating: 0 failure(s), 2 warning(s) on 15 files.
```

The issue is raised for a role "zoning", which has these task names:

```
- name: "Debug Partition initiator WWPNS if verbose"
- name: "[SAR] Retrieve canister-feature-port-wwpn map"
- name: "Debug SAR canister-feature-port-wwpn map if verbose"
- name: "Set fact 'missing ports'"
- name: "Check if canister-feature-port-wwpn map contains all required ports"
- name: "Fail if a required port is missing"
- name: "Set zone-alias array"
- name: "[SAN1] Ensure alias ALIAS_{{ endpoints[context.cpc].SAR }}_01 exists for C1F1P1,C2F2P1 on SAN1"
- name: "[SAN1] Ensure alias ALIAS_{{ endpoints[context.cpc].SAR }}_02 exists for C1F2P1,C2F1P1 on SAN1"
- name: "[SAN2] Ensure alias ALIAS_{{ endpoints[context.cpc].SAR }}_01 exists for C1F1P2,C2F2P2 on SAN2"
- name: "[SAN2] Ensure alias ALIAS_{{ endpoints[context.cpc].SAR }}_02 exists for C1F2P2,C2F1P2 on SAN2"
- name: "[SAN1] Ensure the 8 zones ZONE_{{ params.name | regex_replace('-', '_') }}_0x exist on SAN1"
- name: "[SAN2] Ensure the 8 zones ZONE_{{ params.name | regex_replace('-', '_') }}_0x exist on SAN2"
```
This warning is raised only for the last two task names. The other task names that follow the same naming pattern with the square brackets are not raised as warnings. Other roles used on the playbook and other tasks directly in the playbook also use the naming pattern with the square brackets without the warning being raised.

The issue with the warning is that it is one of those warnings that turns into an error when using `-s` (strict mode), so it prevents us from using strict mode.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.9.1 using ansible 2.13.7
WARNING: ansible-lint is no longer tested under Python 3.8 and will soon require 3.9. Do not report bugs for this version.
```

```
$ cat .ansible-lint 
# Config file for ansible-lint
# Documentation: https://ansible-lint.readthedocs.io/configuring/
# Rules: https://ansible-lint.readthedocs.io/rules/

# Exclude certain directories from being checked.
# Relative directories are relative to this file's location.
exclude_paths:
  - .cache/  # implicit unless exclude_paths is defined in config
  - .github/
  - .pytest_cache/
  - build/
  - DCAutomation.egg-info/
  - endpoint_credentials/
  - playbooks/collections/
  - try/
  - playbooks/try/

# Offline mode disables installation of collections from requirements.yml
# Enabling offline mode causes a warning before ansible-core 2.14
offline: true

# Pattern for valid variable names, in addition to Ansible own requirements
# such as avoiding Python identifiers. To disable the variable name check,
# add 'var-naming' to skip_list.
var_naming_pattern: "^[A-Za-z_][A-Za-z0-9_]*$"

# Ignore these rules.
skip_list:
  - 204  # Lines should be no longer than 160 chars
  - fqcn[action-core]  # Use FQCN for builtin module actions
  - name[template]  # Jinja templates should only be at the end of ...
  - name[play]  # All plays should be named
  - meta-no-info  # No 'galaxy_info' found

# Enable additional rules that are disabled by default.
# enable_list:

# Treat the following rules as warnings, that are by default errors.
# warn_list:
```

- OS: macOS
- Python: 3.8.15
- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

So far I was not able to generate a small testcase that reproduces the issue.

##### Desired Behavior

Task names that start with non-alphabetic characters (such as `[`) should not raise the "name[casing]" warning.

##### Actual Behavior

See above.

**Comment by SirUli** [comment]
I can add that this is triggered when filters are used in the variable in a task name, irrespective of the filter.

**Comment by isuftin** [comment]
Confirm this is also happening when using pipes in the name.

Fails:
```yaml
---
- name: Rundeck | docker | Install Docker

```

Passes:
```yaml
---
- name: Rundeck - docker - Install Docker
```

**Comment by andy-maier** [comment]
Thanks for fixing this! :-)
